metadata/src/main/resources/common/metadata/BrokerRegistrationChangeRecord.json

```diff
@@ -17,14 +17,16 @@
   "apiKey": 17,
   "type": "metadata",
   "name": "BrokerRegistrationChangeRecord",
-  "validVersions": "0",
+  "validVersions": "0-1",
```

Comment:
Hmm. This version bump is not strictly required since we are protecting the tagged field with the metadata version.

For example, in `PartitionChangeRecord` we added the tagged field `LeaderRecoveryState` without changing the version.

@mumrah @cmccabe Do you have an opinion on this?

Reply:
I actually followed @cmccabe's [suggestion](https://lists.apache.org/thread/lgqzoww99gmjzr0v3y23cdgq5fctfrvw).

Reply:
Okay. I am not sure where or how to document this decision but it would be good to be consistent across change type records.

Reply:
I actually think we should not increase the record version unless there is an incompatible change. To my knowledge, that is what the record version has conveyed historically. However, I don't think there's any _harm_ in increasing it, and I also don't think we need to solve it in this PR. Let's go with Colin's suggestion here and we can continue this discussion on the mailing list or offline